pkg/datapath/linux/node.go

```diff
@@ -458,6 +458,26 @@ func upsertIPsecLog(err error, spec string, loc, rem *net.IPNet, spi uint8) {
 	}
 }
 
+func getLinkLocalIp(family int) (*net.IPNet, error) {
+	link, err := netlink.LinkByName(option.Config.EncryptInterface)
+	if err != nil {
+		return nil, err
+	}
+	addr, err := netlink.AddrList(link, family)
+	if err != nil {
+		return nil, err
+	}
+	return addr[0].IPNet, nil
```

Comment:
Is this deterministic?

Reply:
For extra points: How does IPv4/IPv6 factor into this IP selection logic?

Reply:
Using IPv6 tunnel addresses has been carried around with limited testing and as far as I know no users. If we want IPv6 tunnels we probably need a feature request and more testing.

Why addr[0] is the correct address.  The IP is assigned to the interface and _should_ be routable. If user assigns a non-routable IP to the interface, then we have a problem yes. IMO we want a `cilium encryption status` command that dumps all the relevant info into a usable format so we can get visibility here and if it fails we will need user to somehow tell us what to do. I'm not sure we can otherwise automatically pick IPs here. With that in mind addr[0] seems the most logical choice to me.

Reply:
Discussed offline, IPv6 underlay for encryption likely needs some dedicated effort. Then for IPv4 it's relatively uncommon to have multiple addresses and in particular multiple addresses that would have different connectivity such that selecting the wrong one would cause connectivity issues.

OK to pass on this for now, as part of #14638 we can look at exposing visibility into the local IP selection to assist debugging of any potential issues in this area.